src/context/AuthContext.tsx

```diff
@@ -0,0 +1,36 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { jwtDecode } from "jwt-decode";
+import { createContext, useEffect, useState } from "react";
+import {
+  AuthContextProviderProps,
+  AuthContextType,
+  LoginData,
+} from "../interface/UserInfo/UserInfoResponse";
+
+export const AuthContext = createContext<AuthContextType>(null!);
+
+export default function AuthContextProvider({
+  children,
+}: AuthContextProviderProps) {
+  const [loginData, setLoginData] = useState< LoginData | null>(null);
+
+  const saveLoginData = () => {
+    const enCodedToken: any = localStorage.getItem("token");
+    const deCodedToken: LoginData = jwtDecode<LoginData>(enCodedToken);
```

Comment:
I suggest after the login api request that we use the token and retrive the user data from the backend.
![image](https://github.com/user-attachments/assets/5f2c903a-deba-4f6e-be00-2e2fa14fecdd)

This will eliminate the need to use third party library like jwtDecode. The response will also contain the user's profile which we need in the nav bar.

Reply:
The dashboard (navigation bar) is assigned for another teammate